File: irt_gui_4/src/main/java/irt/gui/web/beans/flash/FlashAnswer.java
package irt.gui.web.beans.flash;

import java.util.Arrays;
import java.util.Optional;

import javax.xml.bind.DatatypeConverter;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor @Getter
public enum FlashAnswer {
	UNKNOWN	((byte) -1),
	NULL	((byte) 0),
	ACK		((byte) 0x79),
	NACK	((byte) 0x1F);

	private final Byte answer;

	public static Optional<FlashAnswer> valueOf(byte key){
		final Optional<FlashAnswer> findAny = Arrays
				.stream(values())
				.filter(a->a.answer==key)
				.findAny();
		return findAny;
	}

	@Override
	public String toString() {
		return name()+" (0x"+ DatatypeConverter.printHexBinary(new byte[] {answer}) + ")";
	}

	public boolean match(Byte answer) {
		return Optional.ofNullable(answer).filter(this.answer::equals).isPresent();
	}
}

File: irt_gui_4/src/main/resources/static/js/flash/flash-answer.js
export default class FlashAnswer{
	static names = ['NULL', 'ACK', 'NACK'];
	static code = [0, 0x79, 0x1f];

	constructor(code){
		this.code = code&0xff;
	}

	get name(){
		const index = FlashAnswer.code.indexOf(this.code);
		if(index>=0)
			return FlashAnswer.names[index];

		return 'UNKNOWN';
	}
}